### ISSUE TITLE
Weights can't be saved successfully?

### ISSUE THREAD

**Issue by xfmao** [petrosgk/Kaggle-Carvana-Image-Masking-Challenge#11]
log said metric `val_dice_loss` is not available
my keras version is 2.08 ,using TF backend, TF-gpu version 1.3.0, here are logs

`
Epoch 1/100
/usr/local/lib/python2.7/dist-packages/keras/callbacks.py:496: RuntimeWarning: Early stopping conditioned on metric `val_dice_loss` which is not available. Available metrics are: dice_coeff,loss,val_loss,val_dice_coeff
  (self.monitor, ','.join(list(logs.keys()))), RuntimeWarning
/usr/local/lib/python2.7/dist-packages/keras/callbacks.py:889: RuntimeWarning: Reduce LR on plateau conditioned on metric `val_dice_loss` which is not available. Available metrics are: dice_coeff,loss,lr,val_loss,val_dice_coeff
  (self.monitor, ','.join(list(logs.keys()))), RuntimeWarning
/usr/local/lib/python2.7/dist-packages/keras/callbacks.py:405: RuntimeWarning: Can save best model only with val_dice_loss available, skipping.
  'skipping.' % (self.monitor), RuntimeWarning)
113s - loss: 0.1976 - dice_coeff: 0.8820 - val_loss: 1.7334 - val_dice_coeff: 0.0515
Epoch 2/100
108s - loss: 0.0844 - dice_coeff: 0.9491 - val_loss: 0.0929 - val_dice_coeff: 0.9475
Epoch 3/100
109s - loss: 0.0561 - dice_coeff: 0.9678 - val_loss: 0.0456 - val_dice_coeff: 0.9742

`

**Comment by xfmao** [comment]
It seems monitor can't get 'val_dice_loss' value so ‘Early stopping’，‘Reduce LR on plateau’ ， ‘ModelCheckpoint’ are skipped

**Comment by mgooty** [comment]
This has been fixed [here](https://github.com/petrosgk/Kaggle-Carvana-Image-Masking-Challenge/commit/a32ca525f313037db5a14c56bd6987f6173d29c4)

**Comment by petrosgk** [comment]
#10.
